watch.go

```diff
@@ -103,48 +121,54 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
 
-			var evt []byte
-			var dat []byte
-			isPrefix := true
-			var result []byte
-
-			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
-			// we need bufio#ReadLine()
-			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
-			// so we have to add it to our result buffer.
-			evt, isPrefix, scanErr = scanner.ReadLine()
+			payload := make([]string, 2)
+			scanErr = func(payload []string) error {
```

Comment:
> It is indeed an option. The closure is for me a way to avoid the "break"
with a label. which I don't really like; but this is probably a question of
personal style.

If the case is to avoid having to break a label, can you extract this into it's own function then?

> The other benefits of my solution is that it handles correctly the case
where a reply has more than 2 lines. Perhaps this is what causes the
confusion.

Would you mind giving me an example of a string that would have more than two lines that would not be handled correctly by both pieces of code?